MigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated_withRestart failure

*master* (commit 01267ef5e1bf5433140f10479bbed30d13f6800a)

Failed on: https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-ZuluJDK11/11/testReport/junit/com.hazelcast.internal.partition/MigrationCorrectnessTest/testPartitionData_whenBackupNodesStartedTerminated_withRestart_backups_3_nodes_4_fragmented_false_/

Stacktrace:
```
java.lang.AssertionError: 
Replica: 1 is not found in Partition [42]{
	0:[127.0.0.1]:5702
	2:[127.0.0.1]:5706
	3:[127.0.0.1]:5705
}
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertNotNull(Assert.java:712)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertPartitionVersionsAndBackupValues(PartitionCorrectnessTestSupport.java:269)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertSizeAndData(PartitionCorrectnessTestSupport.java:220)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.access$000(PartitionCorrectnessTestSupport.java:61)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport$AssertSizeAndDataTask.run(PartitionCorrectnessTestSupport.java:359)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1294)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1320)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertSizeAndDataEventually(PartitionCorrectnessTestSupport.java:178)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.assertSizeAndDataEventually(PartitionCorrectnessTestSupport.java:174)
	at com.hazelcast.internal.partition.AbstractMigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated_withRestart(AbstractMigrationCorrectnessTest.java:126)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Standard output: 
```
Hiccups measured while running test 'testPartitionData_whenBackupNodesStartedTerminated_withRestart[backups:3,nodes:4,fragmented:false](com.hazelcast.internal.partition.MigrationCorrectnessTest):'
19:59:00, accumulated pauses: 891 ms, max pause: 196 ms, pauses over 1000 ms: 0
19:59:05, accumulated pauses: 487 ms, max pause: 209 ms, pauses over 1000 ms: 0
19:59:10, accumulated pauses: 115 ms, max pause: 44 ms, pauses over 1000 ms: 0
19:59:15, accumulated pauses: 47 ms, max pause: 2 ms, pauses over 1000 ms: 0
19:59:20, accumulated pauses: 72 ms, max pause: 19 ms, pauses over 1000 ms: 0
19:59:25, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
19:59:30, accumulated pauses: 190 ms, max pause: 145 ms, pauses over 1000 ms: 0
19:59:35, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
19:59:40, accumulated pauses: 48 ms, max pause: 2 ms, pauses over 1000 ms: 0
19:59:45, accumulated pauses: 47 ms, max pause: 0 ms, pauses over 1000 ms: 0
19:59:50, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
19:59:55, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:00, accumulated pauses: 133 ms, max pause: 85 ms, pauses over 1000 ms: 0
20:00:05, accumulated pauses: 51 ms, max pause: 5 ms, pauses over 1000 ms: 0
20:00:10, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:15, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:20, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:25, accumulated pauses: 49 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:00:30, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:35, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:40, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:45, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:00:50, accumulated pauses: 48 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:00:55, accumulated pauses: 52 ms, max pause: 1 ms, pauses over 1000 ms: 0
20:01:00, accumulated pauses: 49 ms, max pause: 1 ms, pauses over 1000 ms: 0
20:01:05, accumulated pauses: 762 ms, max pause: 720 ms, pauses over 1000 ms: 0
20:01:10, accumulated pauses: 1 ms, max pause: 0 ms, pauses over 1000 ms: 0
```

There is extra logging enabled, you can see it in the test output.

--- a/hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java
+++ b/hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationManager.java
@@ -319,9 +319,7 @@ public class MigrationManager {
     void scheduleActiveMigrationFinalization(final MigrationInfo migrationInfo) {
         partitionServiceLock.lock();
         try {
-            // we use activeMigrationInfo because it contains migrated replica fragment namespaces
-            final MigrationInfo activeMigrationInfo = this.activeMigrationInfo;
-            if (activeMigrationInfo != null && migrationInfo.equals(activeMigrationInfo)) {
+            if (migrationInfo.equals(activeMigrationInfo)) {
                 if (activeMigrationInfo.startProcessing()) {
                     activeMigrationInfo.setStatus(migrationInfo.getStatus());
                     finalizeMigration(activeMigrationInfo);
@@ -331,9 +329,9 @@ public class MigrationManager {
                     nodeEngine.getExecutionService().schedule(new Runnable() {
                         @Override
                         public void run() {
-                            scheduleActiveMigrationFinalization(activeMigrationInfo);
+                            scheduleActiveMigrationFinalization(migrationInfo);
                         }
-                    }, 3, TimeUnit.SECONDS);
+                    }, 1, TimeUnit.SECONDS);
                 }
                 return;
             }

